devutils/git-hooks/pre-commit

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+##
+# name: pre-commit-hook
+# author: Axel Martin (for PowerSpikeGG)
+# brief: Ensures commit respect PowerSpikeGG policy
+#
+
+source "$(dirname "${BASH_SOURCE[0]}")/../lib/bash/bazel.bash"
+source "$(dirname "${BASH_SOURCE[0]}")/../lib/bash/logging.bash"
+source "$(dirname "${BASH_SOURCE[0]}")/../lib/bash/prompt.bash"
+
+# Rules checking Go code validity
+PreCommit::GoChecks () {
+    [ -z "$GOPATH" ] && Log::Failure "please install golang on workstation (\$GOPATH missing)"
+
+    # TODO(funkysayu) add go precommits
+}
+
+
+# Rules checking BUILD files validity
+PreCommit::BuildChecks () {
+    local filepath=$1
+
+    [ -z "$GOPATH" ] && Log::Failure "please install golang on workstation (\$GOPATH missing)"
```

Comment:
Initially, GoChecks was an example showing how to make checks on go files. I left it so we may add further elements later. BUILD and Go should be fully separated because we are supposed to use GoChecks by giving a filepath in argument. If we consider GoChecks as a black box (input filepath output status), it does not make sense the use GoChecks in the BuildChecks.